autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/ScalingPolicyTargetCalculator.java

```diff
@@ -40,8 +40,6 @@ public Integer getDesiredAbsoluteNodeCount(ScalingEvent event, int hostGroupNode
                 desiredAbsoluteHostGroupNodeCount = hostGroupNodeCount;
         }
         int minSize = ScalingConstants.DEFAULT_HOSTGROUP_MIN_SIZE;
-        int maxSize = ScalingConstants.DEFAULT_HOSTGROUP_MAX_SIZE;
-
-        return desiredAbsoluteHostGroupNodeCount < minSize ? minSize : desiredAbsoluteHostGroupNodeCount > maxSize ? maxSize : desiredAbsoluteHostGroupNodeCount;
+        return desiredAbsoluteHostGroupNodeCount < minSize ? minSize : desiredAbsoluteHostGroupNodeCount;
```

Comment:
Although unlikely, should this be checking the current limit as well (in case it has been reduced from the time the policy was registered)?
This is also a behaviour question ... in case such a reduction were to happen, what should the user experience look like? - Fail all scaling operations which would otherwise hit the limit OR go to the current limit?

Reply:
Current CB supported max limit is checked just before triggering scale-up in ScalingRequest. 
Considering we are requesting partial scale-ups in CB, it would be better to adjust it to max allowed than fail.
For Ex : Scale-up is requested for 100 nodes when cluster size is 302, failing 98 node scale-up just because 2 nodes  would exceed limit, does not seem correct and also makes Autoscaling less effective.